cmd/createorupdate/createorupdate.go

```diff
@@ -186,10 +240,56 @@ func (s *server) reply(w http.ResponseWriter, req *http.Request) {
 	w.Write(res)
 }
 
+var (
+	method  = flag.String("request", http.MethodPut, "Specify request to send to the OpenShift resource provider. Supported methods are PUT and DELETE.")
+	useProd = flag.Bool("use-prod", false, "If true, send the request to the production OpenShift resource provider.")
+)
+
+func validate() error {
+	switch strings.ToUpper(*method) {
+	case http.MethodPut, http.MethodDelete:
+	default:
+		return fmt.Errorf("invalid request: %s, Supported methods are PUT and DELETE", strings.ToUpper(*method))
+	}
+	return nil
+}
+
 func main() {
+	flag.Parse()
+	if err := validate(); err != nil {
+		logrus.Fatal(err)
+	}
+
 	// simulate the RP
-	s := newServer(os.Getenv("RESOURCEGROUP"))
-	go s.ListenAndServe()
+	if !*useProd {
```

Comment:
this looks little bit unnecessary at this point in time, considering we set address to fakeRpAddr below?

Reply:
No, this is needed to start the fake RP. The address below is set in the client.